types/react/index.d.ts

```diff
@@ -184,6 +184,11 @@ declare namespace React {
     function createFactory<P>(type: ComponentClass<P>): Factory<P>;
 
     // DOM Elements
+    // TODO: generalize this to everything in `keyof ReactHTML`, not just "input"
+    function createElement(
+        type: "input",
+        props?: InputHTMLAttributes<HTMLInputElement> & ClassAttributes<HTMLInputElement>,
+        ...children: ReactNode[]): DetailedReactHTMLElement<{}, HTMLInputElement>;
```

Comment:
Should be `DetailedReactHTMLElement<InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>`, I think.